### PR TITLE
Fix Run Empty Alter Table Add Constraint SQL Error

### LINKED ISSUE #3455: Run empty alter table add constraint sql Error
Please answer these questions before submitting your issue. Thanks!

1. What did you do?
If possible, provide a recipe for reproducing the error.
Run SQL 
```
use test;
create table t (c varchar(30));
ALTER TABLE `test`.`t` ADD INDEX `index1` ();
```
or Run SQL
```
use test;
create table t (c varchar(30));
ALTER TABLE `test`.`t`  ADD UNIQUE INDEX `index1` ();
```

The server will then stop and an error will occur: 

```
reorg.go:73: [info] [ddl] run reorg job done 
panic: runtime error: index out of range

goroutine 63 [running]:
github.com/pingcap/tidb/ddl.(*ddl).onCreateIndex(0xc4206c81c0, 0xc4206c1ca8, 0xc4200ca4d0, 0x1, 0x1, 0x0)
	/home/jenkins/workspace/build_tidb_master/go/src/github.com/pingcap/tidb/ddl/index.go:269 +0x78b
github.com/pingcap/tidb/ddl.(*ddl).runDDLJob(0xc4206c81c0, 0xc4206c1ca8, 0xc4200ca4d0, 0x0)
	/home/jenkins/workspace/build_tidb_master/go/src/github.com/pingcap/tidb/ddl/ddl_worker.go:298 +0x51a
github.com/pingcap/tidb/ddl.(*ddl).handleDDLJobQueue.func1(0x17675c0, 0xc420298a80, 0xc420298a80, 0x0)
	/home/jenkins/workspace/build_tidb_master/go/src/github.com/pingcap/tidb/ddl/ddl_worker.go:230 +0x31c
github.com/pingcap/tidb/kv.RunInNewTxn(0x175cae0, 0xc4206cc240, 0x455500, 0xc4206c1e48, 0xc420022df0, 0xc420022e30)
	/home/jenkins/workspace/build_tidb_master/go/src/github.com/pingcap/tidb/kv/txn.go:43 +0xce
github.com/pingcap/tidb/ddl.(*ddl).handleDDLJobQueue(0xc4206c81c0, 0xc420057500, 0x0)
	/home/jenkins/workspace/build_tidb_master/go/src/github.com/pingcap/tidb/ddl/ddl_worker.go:238 +0xff
github.com/pingcap/tidb/ddl.(*ddl).onDDLWorker(0xc4206c81c0)
	/home/jenkins/workspace/build_tidb_master/go/src/github.com/pingcap/tidb/ddl/ddl_worker.go:58 +0x21e
created by github.com/pingcap/tidb/ddl.(*ddl).start
	/home/jenkins/workspace/build_tidb_master/go/src/github.com/pingcap/tidb/ddl/ddl.go:323 +0xea
```
Then every reboot is the case.

2. What did you expect to see?
ERROR 1210 (HY000): Incorrect arguments to EXECUTE

3. What did you see instead?
ERROR 2013 (HY000): Lost connection to MySQL server during query

4. What version of TiDB are you using (`tidb-server -V`)?

Git Commit Hash: 071281ff91dcd6c35ed36241537aaa6e8822ff8a
UTC Build Time:  2017-06-10 10:51:03

### PR DESCRIPTION
**Pull request by dawxy**
Fix #3455

### CODE REVIEW

**Comment by coocood** [comment]
Thank you, please add test case for it.

**Comment by shenli** [comment]
LGTM Thanks! @dawxy

**Comment by coocood** [comment]
@dawxy 
MySQ returns syntax error instead of Wrong Argument.
Please try to find a solution in parser.

**Comment by coocood** [comment]
@dawxy 
There is a simpler solution, just removed the empty item in `IndexColNameList`.

**Comment by dawxy** [comment]
@coocood 
thank you, I know now.I'll submit it again

**Comment by coocood** [comment]
LGTM

File: parser/parser_test.go [hunk 1]
```diff
@@ -1304,6 +1304,12 @@ func (s *testParserSuite) TestDDL(c *C) {
 		// for truncate statement
 		{"TRUNCATE TABLE t1", true},
 		{"TRUNCATE t1", true},
+
+		//for empty alert table index
```
Comment (zimulala) [inline parser/parser_test.go]: Add a space after the slash.

Comment (dawxy) [inline parser/parser_test.go]: @zimulala 
Well, it has been added.

**Comment by zimulala** [comment]
LGTM

**Pull request merged**